addons/base_automation/static/src/js/base_automation_error_dialog.js

```diff
@@ -1,78 +1,60 @@
+        this.orm = useService("orm");
+    }
 
-        //--------------------------------------------------------------------------
-        // Handlers
-        //--------------------------------------------------------------------------
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
 
-        /**
-        * This method is called when the user clicks on the 'Disable action' button
-        * displayed when a crash occurs in the evaluation of an automated action.
-        * Then, we write `active` to `False` on the automated action to disable it.
-        *
-        * @private
-        * @param {MouseEvent} ev
-        */
-        _onDisableAction: function (ev) {
-            ev.preventDefault();
-            this._rpc({
-                model: 'base.automation',
-                method: 'write',
-                args: [[this.base_automation.id], {
-                    active: false,
-                }],
-            }).then(this.destroy.bind(this));
-        },
-        /**
-        * This method is called when the user clicks on the 'Edit action' button
-        * displayed when a crash occurs in the evaluation of an automated action.
-        * Then, we redirect the user to the automated action form.
-        *
-        * @private
-        * @param {MouseEvent} ev
-        */
-        _onEditAction: function (ev) {
-            ev.preventDefault();
-            this.do_action({
-                name: 'Automated Actions',
-                res_model: 'base.automation',
-                res_id: this.base_automation.id,
-                views: [[false, 'form']],
-                type: 'ir.actions.act_window',
-                view_mode: 'form',
-            });
-        },
-    });
+    /**
+     * This method is called when the user clicks on the 'Disable action' button
+     * displayed when a crash occurs in the evaluation of an automated action.
+     * Then, we write `active` to `False` on the automated action to disable it.
+     *
+     * @private
+     * @param {MouseEvent} ev
+     */
+    async disableAction(ev) {
+        await this.orm.write("base.automation", [this.actionId], {
+            active: false,
+        });
+        this.close();
+    }
+    /**
+     * This method is called when the user clicks on the 'Edit action' button
+     * displayed when a crash occurs in the evaluation of an automated action.
+     * Then, we redirect the user to the automated action form.
+     *
+     * @private
+     * @param {MouseEvent} ev
+     */
+    editAction(ev) {
+        this.actionService.doAction({
+            name: "Automated Actions",
+            res_model: "base.automation",
+            res_id: this.actionId,
+            views: [[false, "form"]],
+            type: "ir.actions.act_window",
+            view_mode: "form",
+        });
+        this.close();
```

Comment:
doesn't close automatically when executing another action?

Reply:
Just checked: it doesn't.